framework/include/utils/InputParametersChecksUtils.h

```diff
@@ -371,9 +371,13 @@ InputParametersChecksUtils<C>::checkVectorParamsNoOverlap(
       {
         auto copy_params = param_vec;
         copy_params.erase(std::find(copy_params.begin(), copy_params.end(), param));
-        forwardMooseError("Item '" + value + "' specified in vector parameter '" + param +
-                          "' is also present in one or more of the parameters '" +
-                          Moose::stringify(copy_params) + "', which is not allowed.");
+        if (copy_params.size())
+          forwardMooseError("Item '" + value + "' specified in vector parameter '" + param +
+                            "' is also present in one or more of the parameters '" +
+                            Moose::stringify(copy_params) + "', which is not allowed.");
+        else
+          forwardMooseError("Item '" + value + "' specified in vector parameter '" + param +
+                            "' is repeated, which is not allowed.");
```

Comment:
I think I finally figured out this logic, but it required a few minutes of thinking. A code comment would be good, like "check whether there are repeated values within one object's parameter or whether the repeat is due to an overlap in one object's parameter value and another object's same parameter value"

Reply:
done